File: src/main/java/my/maryvk/maryvkweb/domain/RelationChange.java
package my.maryvk.maryvkweb.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.LocalDateTime;

@Entity
@Data @Builder @AllArgsConstructor @NoArgsConstructor
public class RelationChange {

    @Id @GeneratedValue
    private Long id;

    private LocalDateTime time;
    private Integer ownerId;
    private Integer targetId;

    private RelationType relationType;
    private Boolean isAppeared;
}

File: src/main/java/my/maryvk/maryvkweb/domain/RelationChange.kt
package my.maryvk.maryvkweb.domain

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity data class RelationChange(
        @Id @GeneratedValue var id: Long?,

        var time: LocalDateTime?,
        var ownerId: Int?,
        var targetId: Int?,

        var relationType: RelationType?,
        var isAppeared: Boolean?
) {
    constructor() : this(null, null, null, null, null, null)
}